src/DecryptionFailureTracker.js

```diff
@@ -163,7 +162,8 @@ export default class DecryptionFailureTracker {
      */
     trackFailure() {
         if (this.failuresToTrack.length > 0) {
-            this.trackDecryptionFailure(this.failuresToTrack.shift());
+            // Remove all failures, and expose the number of failures
+            this.trackDecryptionFailure(this.failuresToTrack.splice(0).length);
```

Comment:
Even with the comment, this was a bit confusing until I realised the tracker really does just want a total number of failed decryptions, especially as this class has been carefully tracking each failure up until this point where it throws them away. Maybe just expand on the comment a little?

Reply:
Fair enough! Shall do.

Reply:
> this class has been carefully tracking each failure up until this point where it throws them away

It still has a valid reason to do this, so I'm not quite sure what you're confused about.

Not sure what you mean by "throwing them away", or why there's a problem with doing this. I'll make it clearer that there's still an intention to track by error type later.

Reply:
Well, by 'throwing them away' I'm just referring to the fact that splice(0) empties the array, returns a new one which we then just take the length of and do nothing else with. Since this is what the interface wants, it's actually correct, it was just a bit confusing to read and work out what was going on. The comment helps though.